docs/reference/clients/python/index.rst

```diff
@@ -4,51 +4,214 @@
 
-* :ref:`gel-python-codegen`
+.. note::
 
-  Python code generation command-line tool documentation.
+  If you're using |Gel| Cloud to host your development instance, you can use the :gelcmd:`cloud login` command to authenticate with |Gel| Cloud and then use the :gelcmd:`project init --server-instance <instance-name>` command to create a local project-linked instance that is linked to an Gel Cloud instance. For more details, see :ref:`the Gel Cloud guide <ref_guide_cloud>`.
 
-* :ref:`gel-python-advanced`
+.. tabs::
 
-  Advanced usages of the state and optional customization.
+  .. code-tab:: python
+    :caption: Blocking
 
+    import datetime
+    import gel
 
-.. toctree::
-   :maxdepth: 3
-   :hidden:
+    client = gel.create_client()
+
+    client.query("""
+        INSERT User {
+            name := <str>$name,
+            dob := <cal::local_date>$dob
+        }
+    """, name="Bob", dob=datetime.date(1984, 3, 1))
+
+    user_set = client.query(
+        "SELECT User {name, dob} FILTER .name = <str>$name", name="Bob")
+    # *user_set* now contains
+    # Set{Object{name := 'Bob', dob := datetime.date(1984, 3, 1)}}
+
+    client.close()
+
+  .. code-tab:: python
+    :caption: AsyncIO
+
+    import asyncio
+    import datetime
+    import gel
+
+    client = gel.create_async_client()
+
+    async def main():
+        await client.query("""
+            INSERT User {
+                name := <str>$name,
+                dob := <cal::local_date>$dob
+            }
+        """, name="Bob", dob=datetime.date(1984, 3, 1))
+
+        user_set = await client.query(
+            "SELECT User {name, dob} FILTER .name = <str>$name", name="Bob")
+        # *user_set* now contains
+        # Set{Object{name := 'Bob', dob := datetime.date(1984, 3, 1)}}
+
+        await client.aclose()
+
+    asyncio.run(main())
+
+.. _gel-python-connection-pool:
+
+Client connection pools
+-----------------------
+
+For server-type applications that handle frequent requests and need the database connection for a short period of time while handling a request, you will want to use a connection pool. Both :py:class:`gel.Client` and :py:class:`gel.AsyncIOClient` come with such a pool.
+
+For :py:class:`gel.Client`, all methods are thread-safe. You can share the same client instance safely across multiple threads, and run queries concurrently. Likewise, :py:class:`~gel.AsyncIOClient` is designed to be shared among different :py:class:`asyncio.Task`/coroutines for concurrency.
+
+Below is an example of a web API server running `fastapi <https://fastapi.tiangolo.com/>`_:
```

Comment:
Someone please double-check my work here. I mostly did the aiohttp to fastapi conversion on vibes.